common/src/main/java/io/novafoundation/nova/common/utils/recyclerView/space/SpaceBetween.kt

```diff
@@ -0,0 +1,37 @@
+package io.novafoundation.nova.common.utils.recyclerView.space
+
+import android.graphics.Rect
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import io.novafoundation.nova.common.utils.dp
+import io.novafoundation.nova.common.utils.recyclerView.ViewType
+
+class SpaceBetween(
+    private val upperViewType: ViewType,
+    private val lowerViewType: ViewType,
+    private val spaceDp: Int
+) : RecyclerViewItemSpace {
+
+    override fun handleSpace(outRect: Rect, view: View, parent: RecyclerView): Boolean {
+        if (shouldSetSpaceForItems(parent, view)) {
+            setSpaceBetweenItems(outRect, view)
+            return true
+        }
+
+        return false
+    }
+
+    private fun shouldSetSpaceForItems(
+        parent: RecyclerView,
+        view: View
+    ): Boolean {
+        parent.getViewType(view).takeIf { it == upperViewType } ?: return false
+        parent.getNextViewType(view).takeIf { it == lowerViewType } ?: return false
```

Comment:
I think its better to explicitly include non-null check here - it took me two attempts to understand we use `takeIf` because `getViewType` is nullable
So

`currentViewType != null && currentViewType == upperViewType`